cli/flags.rs

```diff
@@ -1643,10 +1643,16 @@ fn compat_arg<'a, 'b>() -> Arg<'a, 'b> {
 fn watch_arg<'a, 'b>() -> Arg<'a, 'b> {
   Arg::with_name("watch")
     .long("watch")
+    .value_name("FILES")
```

Comment:
It seems we need to make this value conditional because it's not supported for certain commands? That will also allow us to remove some of the logic in `watch_arg_parse` and have it error when someone provides this for subcommands that don't support it.

Reply:
@bartlomieju and I briefly discussed this here: https://github.com/denoland/deno/pull/13087#discussion_r769138224
I'm not really sure what would be the best way to do this. Should `watch_arg_parse` return a `Result<>` with every call to it manually handling the error? Or is there code somewhere that I can more elegantly reuse for handling argument errors?
Also, is there anything that needs to change to the `Arg::with_name` parameters to let the argument handler know it only supports a value under certain circumstances?

Reply:
It would be how the argument is defined here. So something like the following:

```rs
fn watch_arg<'a, 'b>(takes_files: bool) -> Arg<'a, 'b> {
  let arg = Arg::with_name("watch")
    .long("watch")
    .help("UNSTABLE: Watch for file changes and restart process automatically");
  if takes_files {
    arg.value_name("FILES")
      .min_values(0)
      .takes_value(true)
      .use_delimiter(true)
      .require_equals(true)
      .long_help(...goeshere...)
  } else {
    arg.long_help(...goeshere....)
  }
}
```

Reply:
Ah interesting, I wasn't aware arguments could contain logic like this. Thanks I'll try this!

Reply:
@jespertheend let me know if you want to tackle that, if not I can fixup and land the PR.

Reply:
I'm working on it atm. I'll have a patch ready in a bit.

Reply:
It seems removing `takes_value(true)` doesn't actually cause any errors when invoked with `--watch=somefile.txt`.
So all the new patch really does is update `--help` accordingly I believe.

Reply:
Yeah, that's ok. As long as the help text is good now 👍